### PR TITLE
add nodejs/npmmirror.lua to support use cdn.npmmirror.com to speed up nodejs install

### PR DESCRIPTION
**Pull request by yimiaoxiehou**
add nodejs/npmmirror.lua to support use cdn.npmmirror.com to speed up nodejs install

### CODE REVIEW

File: nodejs/npmmirror.lua [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+NodeBaseUrl = "https://cdn.npmmirror.com/binaries/node/v%s/"
+FileName = "node-v%s-%s-%s%s"
+npmDownloadUrl = "https://cdn.npmmirror.com/binaries/node/npm/v%s.%s"
+
+VersionSourceUrl = "https://cdn.npmmirror.com/binaries/node/index.json"
+
+PLUGIN = {
+    name = "nodejs",
+    author = "Aooohan",
```
Comment (aooohan) [inline nodejs/npmmirror.lua]: This can be changed to your name.

File: nodejs/npmmirror.lua [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    version = "0.0.2",
```
Comment (aooohan) [inline nodejs/npmmirror.lua]: For new plugins. The initial version number should be 0.0.1.
```suggestion
    version = "0.0.1",
```

**Review by aooohan — CHANGES_REQUESTED** [review]

**Comment by aooohan** [comment]
Thanks for your contribution. And I want to hear your opinions about mirroring, as it just so happens that a user has raised a new issue. Which of the two options do you prefer? I want to hear your opinions. ; )

https://github.com/version-fox/vfox/issues/33#issuecomment-1914059836

File: nodejs/npmmirror.lua [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    return result
+end
+
+--- Expansion point
+function PLUGIN:PostInstall(ctx)
+    local rootPath = ctx.rootPath
+    local sdkInfo = ctx.sdkInfo['nodejs']
+    local path = sdkInfo.path
+    local version = sdkInfo.version
+    local name = sdkInfo.name
```
Comment (aooohan) [inline nodejs/npmmirror.lua]: You can delete these lines safely, I just wrote them to test the extension, they don't make sense.

File: nodejs/npmmirror.lua [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+npmDownloadUrl = "https://cdn.npmmirror.com/binaries/node/npm/v%s.%s"
+
+VersionSourceUrl = "https://cdn.npmmirror.com/binaries/node/index.json"
+
+PLUGIN = {
+    name = "nodejs",
+    author = "yimiaoxiehou",
+    version = "0.0.1",
+    description = "install Node.js use https://cdn.npmmirror.com",
+    updateUrl = "https://raw.githubusercontent.com/version-fox/version-fox-plugins/main/node/npmmirror.lua",
```
Comment (aooohan) [inline nodejs/npmmirror.lua]:
```suggestion
    updateUrl = "https://raw.githubusercontent.com/version-fox/version-fox-plugins/main/nodejs/npmmirror.lua",
```

**Review by aooohan — CHANGES_REQUESTED** [review]

**Review by yimiaoxiehou — COMMENTED** [review]
👍

**Review by aooohan — APPROVED** [review]
LGTM.

**Comment by aooohan** [comment]
Thanks for your contribution. This plugin is already available now. ; ) 
image

**Pull request merged**